### ISSUE TITLE
fully static build

### ISSUE THREAD

**Issue by fixerivan** [servo/servo#24649]
is it possible to build servo as a fully static binary using RUST-MUSL https://doc.rust-lang.org/edition-guide/rust-2018/platform-and-target-support/musl-support-for-fully-static-binaries.html ?

out of the box it fails on:

error: cannot produce cdylib for `simpleservo_capi v0.0.1 (/home/vania/servo/ports/libsimpleservo/capi)` as the target `x86_64-unknown-linux-musl` does not support these crate types

**Comment by jdm** [comment]
What is the build command you are running? The simpleservo_capi is only used for embedding, rather than building a single binary.

**Comment by SimonSapin** [comment]
This seems hard, since we compile a number of C and C++ libraries and those would also need to support musl in their respective build systems.

**Comment by fixerivan** [comment]
@jdm  the error was generated by running:

RUSTFLAGS='-L ./support/linux/gstreamer/gst/lib' cargo build --release --target=x86_64-unknown-linux-musl

**Comment by jdm** [comment]
@fixerivan You might have better luck with `--manifest-path ports/glutin/Cargo.toml`.

**Comment by fixerivan** [comment]
@SimonSapin yes, thats why i am trying to compile it on Alpine which is a MUSL based distro 

but i am getting crate-type errors on macros such as:

alpine:~/servo# ./mach build --release --target=x86_64-unknown-linux-musl
rustc 1.40.0-nightly (246be7e1a 2019-10-25)
info: component 'rust-std' for target 'x86_64-unknown-linux-musl' is up to date
error: cannot produce proc-macro for `cssparser-macros v0.6.0` as the target `x86_64-unknown-linux-musl` does not support these crate types
Build FAILED in 0:00:00

any ideas? thx

**Comment by fixerivan** [comment]
@jdm thx for the tip but the result is the same:

alpine:~/servo# cargo clean
alpine:~/servo# ./mach clean
Removing virtualenv directory: /root/servo/python/_virtualenv
alpine:~/servo# cargo build --manifest-path ports/glutin/Cargo.toml
error: cannot produce proc-macro for `cssparser-macros v0.6.0` as the target `x86_64-unknown-linux-musl` does not support these crate types

**Comment by SimonSapin** [comment]
Honestly, I think this error is just the beginning. I expect pushing this project to completion would take considerable effort at best, including a lot of investigation of unclear error messages. I’m personally not interested in investing that effort myself.
